kernel/kernel-api/src/main/java/io/delta/kernel/expressions/ScalarExpression.java

```diff
@@ -73,4 +80,11 @@ public String getName() {
     public List<Expression> getChildren() {
         return children;
     }
+
+    public DataType getOutputType() {
+        if (this.outputType == null) {
```

Comment:
This might not be the best idea, but we need to propagate outputType to avoid calculating leas common type multiple times, should we make all ScalarExpression have outputType?